backend/src/agent/persistence/client/pipeline_client.go

```diff
@@ -55,8 +55,8 @@ func NewPipelineClient(
 	mlPipelineServiceName string,
 	mlPipelineServiceHttpPort string,
 	mlPipelineServiceGRPCPort string) (*PipelineClient, error) {
-	httpAddress := fmt.Sprintf(addressTemp, mlPipelineServiceName, namespace, mlPipelineServiceHttpPort)
```

Comment:
instead, can you parameterize the domain name, and give default value "local"?
%s.%s.svc.cluster.%s:%s

It would be easier for customizations in future. we don't assume api and persistentagent in the same namespace.

Reply:
We need to put "cluster" together with "local", as this field represents <zone>, https://github.com/kubernetes/dns/blob/master/docs/specification.md#21---definitions. If we go that way, we'll have three configs to generate the full service name: <service-name>.<namespace>.svc.<zone>. Another way (just as the diff) would be, just using one config to config the full path, which includes the three configs aforementioned.

Reply:
i think these deserves flexible configurations so having a "zone" parameter with default value cluster.local make sense. 
With that it's easy to change namespace without changing anything else. 
WDYT?

Reply:
haha, I'm ok with 1 config or 3 configs. Another thing is, we need to introduce another config "informerNamespace" in #901. I used to think about releasing "namespace" config here to replace "informerNamespace", to keep consistent with #851 swf.

Reply:
sounds good. do you want to do it as part of this PR?

Reply:
Updated the diff to remove "namespace" which is not needed anymore.